src/fides/api/schemas/connection_configuration/connection_config.py

```diff
@@ -47,43 +49,41 @@ class Config:
         extra = Extra.forbid
 
 
-class TestStatus(Enum):
-    passed = "passed"
-    failed = "failed"
-    untested = "untested"
-
-    def str_to_bool(self) -> Optional[bool]:
-        """Translates query param string to optional/bool value
-        for filtering ConnectionConfig.last_test_succeeded field"""
-        if self == self.passed:
-            return True
-        if self == self.failed:
-            return False
-        return None
-
-
-class SystemType(Enum):
-    saas = "saas"
-    database = "database"
-    manual = "manual"
-    email = "email"
-
-
-class ConnectionSystemTypeMap(BaseModel):
+def mask_sensitive_fields(
+    connection_secrets: Dict[str, Any], secret_schema: Dict[str, Any]
+) -> Dict[str, Any]:
     """
-    Describes the returned schema for connection types
+    Mask sensitive fields in the given secrets based on the provided schema.
+    This function traverses the given secrets dictionary and uses the provided schema to
+    identify fields that have been marked as sensitive. The function replaces the sensitive
+    field values with a mask string ('********').
+    Note: This function modifies the input dictionary in-place.
```

Comment:
Let's remove this from the docstring since the input isn't being modified in place anymore.